packages/maskbook/src/plugins/Ideamarket/define.tsx

```diff
@@ -0,0 +1,35 @@
+import { PluginConfig, PluginStage, PluginScope } from '../types'
+import { usePostInfoDetails } from '../../components/DataSource/usePostInfo'
+import Fetcher from './Fetcher'
+import { makeStyles } from '@material-ui/core'
+
+const useStyles = makeStyles(() => ({
+    layout: {
+        position: 'absolute',
+        right: '20px',
+        top: 0,
+        marginTop: '-37px',
+    },
+}))
+
+export const IdeamarketPluginDefine: PluginConfig = {
+    pluginName: 'Ideamarket',
+    identifier: 'ideamarket.io',
+    stage: PluginStage.Development,
+    scope: PluginScope.Internal,
+
+    postInspector: function Component(): JSX.Element | null {
+        const user = usePostInfoDetails('postBy').userId
+        const classes = useStyles()
+
+        if (!user) return null
+
+        const username = '@' + user.toLowerCase()
+
+        return (
+            <div className={classes.layout}>
+                <Fetcher username={username} />
```

Comment:
It seems like this plugin will make a request whatever. I'm worrying if it will be the next #2840

Reply:
Yes, you are right. It does make a request for every tweet visible on the viewport. This is because we want to show the Price of the token in line with the tweet without any user interaction. 

@bobface can we do something about this?

Reply:
I don't think so. Our API allows to query multiple listings in a single query, but I don't think there is an easy way to get all Tweets on the screen to make a bundle query.

Reply:
Yeah, our API is generally not optimized for this bundle query case. I think we need a user action to check out the price, or the API query must be very fast and cheap otherwise the API server will down cause too many requests